src/MembershipContextFactory.php

```diff
@@ -4,97 +4,103 @@
-use WMDE\Fundraising\MembershipContext\DataAccess\DoctrineApplicationAuthorizer;
 use WMDE\Fundraising\MembershipContext\DataAccess\DoctrineMembershipApplicationPrePersistSubscriber;
-use WMDE\Fundraising\Store\Factory as StoreFactory;
-use WMDE\Fundraising\Store\Installer;
 
 /**
  * @licence GNU GPL v2+
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
-class MembershipContextFactory implements ServiceProviderInterface {
-
-	private $config;
+class MembershipContextFactory {
 
 	/**
-	 * @var Container
+	 * Use this constant for MappingDriverChain::addDriver
 	 */
-	private $pimple;
+	public const DOCTRINE_ADDITIONAL_ENTITIES = [
+		'WMDE\Fundraising\MembershipContext\DataAccess\DoctrineEntities' => __DIR__ . '/DoctrineEntities'
+	];
+
+	public const ENTITY_PATHS = [
+		__DIR__ . '/DataAccess/DoctrineEntities/'
+	];
+
+	private array $config;
+	private Configuration $doctrineConfig;
 
 	private $addDoctrineSubscribers = true;
 
-	public function __construct( array $config ) {
+	private ?Connection $connection;
+	private ?EntityManager $entityManager;
+	private ?RandomMembershipTokenGenerator $tokenGenerator;
+
+	public function __construct( array $config, Configuration $doctrineConfig ) {
 		$this->config = $config;
-		$this->pimple = $this->newPimple();
-	}
+		$this->doctrineConfig = $doctrineConfig;
 
-	private function newPimple(): Container {
-		$pimple = new Container();
-		$this->register( $pimple );
-		return $pimple;
+		$this->connection = null;
+		$this->entityManager = null;
+		$this->tokenGenerator = null;
 	}
 
-	public function register( Container $container ): void {
-		$container['dbal_connection'] = function() {
-			return DriverManager::getConnection( $this->config['db'] );
-		};
-
-		$container['entity_manager'] = function() {
-			$entityManager = ( new StoreFactory( $this->getConnection(), $this->getVarPath() . '/doctrine_proxies' ) )
-				->getEntityManager();
-			if ( $this->addDoctrineSubscribers ) {
-				$entityManager->getEventManager()->addEventSubscriber(
-					$this->newDoctrineMembershipPrePersistSubscriber()
-				);
-			}
-
-			return $entityManager;
-		};
-
-		$container['fundraising.membership.application.token_generator'] = function() {
-			return new RandomMembershipTokenGenerator(
-				$this->config['token-length'],
-				new \DateInterval( $this->config['token-validity-timestamp'] )
-			);
-		};
-
-		$container['fundraising.membership.application.authorizer.class'] = DoctrineApplicationAuthorizer::class;
-		// @todo Consider if the tokens should not better be method parameters (see ApplicationAuthorizer interface)
-		$container['fundraising.membership.application.authorizer.update_token'] = null;
-		$container['fundraising.membership.application.authorizer.access_token'] = null;
-
-		$container['fundraising.membership.application.authorizer'] = $container->factory( function ( Container $container ): ApplicationAuthorizer {
```

Comment:
@gbirke Do you know what this was for? It didn't seem to be doing anything.

Reply:
This is a place where we don't expose a service via getters of the MembershipContextFactory, but with the Pimple DIC. See https://github.com/wmde/FundraisingFrontend/blob/master/src/Factories/FunFunFactory.php#L1327 it's used there. 

You can remove this here, but we'll have to remember to change the method in FunFunFactory.